integrations/kubecost/integration.py

```diff
@@ -0,0 +1,156 @@
+import re
+from typing import Literal
+
+from pydantic.fields import Field
+
+from port_ocean.core.handlers.port_app_config.api import APIPortAppConfig
+from port_ocean.core.handlers.port_app_config.models import (
+    PortAppConfig,
+    ResourceConfig,
+    Selector,
+)
+from port_ocean.core.integrations.base import BaseIntegration
+
+
+class DatePairField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the date pair value
+        regex = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z,\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid date pair format. Use 'YYYY-MM-DDTHH:MM:SSZ,YYYY-MM-DDTHH:MM:SSZ'."
+            )
+
+
+class UnixtimePairField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the unixtime pair value
+        regex = r"^\d+,\d+$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid unixtime pair format. Use 'X,Y' (X and Y are positive integers)."
+            )
+
+
+class AggregationField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the aggregation value
+        regex = r"^(cluster|node|namespace|controllerKind|controller|service|pod|container|invoiceEntityID|accountID|provider|label:name|annotation:name)(,(cluster|node|namespace|controllerKind|controller|service|pod|container|invoiceEntityID|accountID|provider|label:name|annotation:name))*$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid aggregation format. Use 'cluster', 'node', 'namespace', 'controllerKind', 'controller', 'service', 'pod', 'container', 'label:name', or 'annotation:name'."
+            )
+
+
+class DurationField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the step value
+        regex = r"^\d+[smhd]$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid duration format. Use 'Xs', 'Xm', 'Xh', or 'Xd' (X is a positive integer)."
+            )
+
+
+class KubecostSelector(Selector):
```

Comment:
add a `def generate_params():` function that will generate the query params for the selector
`.dict(exclude_unset=True, by_alias=True)`
dont forget to pop `query` out

then use it instead of all the if statments in the client